operator/builtin/transformer/move/move_test.go

```diff
@@ -0,0 +1,166 @@
+	"github.com/open-telemetry/opentelemetry-log-collection/operator"
+	"github.com/open-telemetry/opentelemetry-log-collection/testutil"
+	"github.com/stretchr/testify/require"
+)
+
+func TestMoveOperator(t *testing.T) {
+	newTestEntry := func() *entry.Entry {
+		e := entry.New()
+		e.Timestamp = time.Unix(1586632809, 0)
+		e.Record = map[string]interface{}{
+			"key": "val",
+			"nested": map[string]interface{}{
+				"nestedkey": "nestedval",
+			},
+		}
+		return e
+	}
+
+	cases := []struct {
+		name      string
+		moveOp    MoveOperator
+		input     *entry.Entry
+		output    *entry.Entry
+		expectErr bool
+	}{
+		{
+			name: "MoveRecordToRecord",
+			moveOp: MoveOperator{
+				From: entry.NewRecordField("key"),
+				To:   entry.NewRecordField("new"),
+			},
+			input: newTestEntry(),
+			output: func() *entry.Entry {
+				e := newTestEntry()
+				e.Record = map[string]interface{}{
+					"new": "val",
+					"nested": map[string]interface{}{
+						"nestedkey": "nestedval",
+					},
+				}
+				return e
+			}(),
+		},
+		{
+			name: "MoveRecordToAttribute",
+			moveOp: MoveOperator{
+				From: entry.NewRecordField("key"),
+				To:   entry.NewAttributeField("new"),
+			},
+			input: newTestEntry(),
+			output: func() *entry.Entry {
+				e := newTestEntry()
+				e.Record = map[string]interface{}{
+					"nested": map[string]interface{}{
+						"nestedkey": "nestedval",
+					},
+				}
+				e.Attributes = map[string]string{"new": "val"}
+				return e
+			}(),
+		},
+		{
+			name: "MoveAttributeToRecord",
+			moveOp: MoveOperator{
+				From: entry.NewAttributeField("new"),
+				To:   entry.NewRecordField("new"),
+			},
+			input: func() *entry.Entry {
+				e := newTestEntry()
+				e.Attributes = map[string]string{"new": "val"}
+				return e
+			}(),
+			output: func() *entry.Entry {
+				e := newTestEntry()
+				e.Record = map[string]interface{}{
+					"key": "val",
+					"new": "val",
+					"nested": map[string]interface{}{
+						"nestedkey": "nestedval",
+					},
+				}
+				e.Attributes = map[string]string{}
+				return e
+			}(),
+		},
+		{
+			name: "MoveAttributeToResource",
+			moveOp: MoveOperator{
+				From: entry.NewAttributeField("new"),
+				To:   entry.NewResourceField("new"),
+			},
+			input: func() *entry.Entry {
+				e := newTestEntry()
+				e.Attributes = map[string]string{"new": "val"}
+				return e
+			}(),
+			output: func() *entry.Entry {
+				e := newTestEntry()
+				e.Resource = map[string]string{"new": "val"}
+				e.Attributes = map[string]string{}
+				return e
+			}(),
+		},
+		{
+			name: "MoveNest",
+			moveOp: MoveOperator{
+				From: entry.NewRecordField("nested"),
+				To:   entry.NewRecordField("NewNested"),
+			},
+			input: newTestEntry(),
+			output: func() *entry.Entry {
+				e := newTestEntry()
+				e.Record = map[string]interface{}{
+					"key": "val",
+					"NewNested": map[string]interface{}{
+						"nestedkey": "nestedval",
+					},
+				}
+				return e
+			}(),
+		},
+	}
```

Comment:
I think we need to cover more cases in these tests. This is a good start, but let's consider more possible field locations, and more types of values. 

Let's consider at least the following:
- `$record.some_string` 
- `$record.some_object` // move the whole object
- `$record.some_object.string` // move nested string
- `$record.some_object.some_object` // move nested object
- `$resource.some_key`
- `$attributes.some_key`

I think we should be able to set up the test so that it runs through all combinations of the above. (It should fail when moving an object to `attributes` or `resource`, but those failures should be tested too).

Reply:
do you mean that nested should fail when moving to attributes or resource?